include/kernel/ebpf_registry_helper.h

```diff
@@ -13,123 +13,43 @@
 #define GUID_STRING_LENGTH 38 // not including the null terminator.
 
 typedef _Return_type_success_(NT_SUCCESS(return )) uint32_t ebpf_registry_result_t;
-
 typedef HANDLE ebpf_registry_key_t;
 
-static void
-close_registry_key(ebpf_registry_key_t key)
-{
-    ZwClose(key);
-}
-
-_Success_(return == 0) static NTSTATUS
-    convert_guid_to_string(_In_ const GUID* guid, _Out_writes_all_(string_size) wchar_t* string, size_t string_size)
-{
-    UNICODE_STRING unicode_string;
-    NTSTATUS status = RtlStringFromGUID(guid, &unicode_string);
-    if (status != STATUS_SUCCESS) {
-        goto Exit;
-    }
-
-    if (string_size < GUID_STRING_LENGTH + 1) {
-        status = STATUS_BUFFER_TOO_SMALL;
-        goto Exit;
-    }
-
-    __analysis_assume(unicode_string.MaximumLength >= GUID_STRING_LENGTH * 2);
-    __analysis_assume(unicode_string.Buffer != NULL);
-    // Copy the buffer to the output string.
-    memcpy(string, unicode_string.Buffer, GUID_STRING_LENGTH * 2);
-    string[GUID_STRING_LENGTH] = L'\0';
-
-Exit:
-    return status;
-}
-
-static _Must_inspect_result_ ebpf_registry_result_t
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Mocks for building platform_kernel & netebpfext when including the common ebpf_store_helper.h in kernel mode.
+// Currently these prototypes only need (existing) implementations in user mode, as they are not referenced in kernel
+// binaries.
+_Success_(return == 0) uint32_t open_registry_key(
+    ebpf_registry_key_t root_key, _In_opt_z_ const wchar_t* sub_key, uint32_t flags, _Out_ ebpf_registry_key_t* key);
+_Must_inspect_result_ ebpf_registry_result_t
+delete_registry_key(ebpf_registry_key_t root_key, _In_z_ const wchar_t* sub_key);
+_Must_inspect_result_ ebpf_registry_result_t
+delete_registry_tree(ebpf_registry_key_t root_key, _In_opt_z_ const wchar_t* sub_key);
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+_Success_(return == 0) NTSTATUS
```

Comment:
Per [this doc](https://learn.microsoft.com/en-us/cpp/code-quality/annotating-function-behavior?view=msvc-170#successfailure-annotations), ```NTSTATUS``` already has a ```__Success__``` notation built into it, so was curious why we needed to re-define it?  If this is the same as the default, it would be redundant and if we're changing it (per that link, re-defining overrides the default) I was curious why the default was insufficient?

Reply:
I believe Dhiren is correct.
```suggestion
NTSTATUS
```

Reply:
Given the scope of this PR (i.e. wrapping direct calls to the registry APIs to store APIs), I filed issue https://github.com/microsoft/ebpf-for-windows/issues/2123 to harden all registry APIs.

Reply:
Umm, I'm not sure I'd agree with this resolution.  This is _new_ code introduced by this PR (and thus within scope) and ought to be fixed at the get go. IMO, merging it only to subsequently change it via a brand new issue seems redundant.  All the more so, since Dave's suggestion changes just a single line.

Is there some other implication making this change that I'm missing?

Reply:
> Umm, I'm not sure I'd agree with this resolution. This is _new_ code introduced by this PR (and thus within scope) and ought to be fixed at the get go. 

Same as [here](https://github.com/microsoft/ebpf-for-windows/pull/2114#discussion_r1113956256)

Reply:
> Per [this doc](https://learn.microsoft.com/en-us/cpp/code-quality/annotating-function-behavior?view=msvc-170#successfailure-annotations), `NTSTATUS` already has a `__Success__` notation built into it, so was curious why we needed to re-define it? If this is the same as the default, it would be redundant and if we're changing it (per that link, re-defining overrides the default) I was curious why the default was insufficient?

Looking this up in `\Windows Kits\10\Include\10.0.22621.0\shared\ntdef.h`, `NTSTATUS` actually resolves to:
```c
typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
```
whereas the current definition wants `(return == 0)`.

Reply:
Why does the current implementation want something other than the Windows standard?

Reply:
Can't tell about the history, but since it's only used in `\include\ebpf_store_helper.h`, et ivi just checked for 0, removing it won't affect the current code.